src/TodoList.tsx

```diff
@@ -0,0 +1,106 @@
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+
+interface TodoListProps {
+  todos: Todo[],
+  removeTodo: React.Dispatch<React.SetStateAction<boolean>>,
+  editTodo: React.Dispatch<React.SetStateAction<boolean>>,
+}
+
+export const TodoList: React.FC<TodoListProps> = ({
+  todos,
+  removeTodo,
+  editTodo,
+}) => {
+  const handleTodoRemoval = () => {
+    removeTodo(true);
+  };
+
+  const handleTodoTitleChange = () => {
+    editTodo(true);
+  };
+
+  return (
+    <section className="todoapp__main">
+      {/* This is a completed todo */}
+
+      {todos.map((todo) => (
+        <div
+          className={classNames('todo', {
+            completed: todo.completed,
+          })}
+          key={todo.id}
+          onDoubleClick={handleTodoTitleChange}
+        >
+          <label className="todo__status-label">
+            <input
+              type="checkbox"
+              className="todo__status"
+              checked
+            />
+          </label>
+
+          <span className="todo__title">{todo.title}</span>
+
+          {/* Remove button appears only on hover */}
+          <button
+            type="button"
+            className="todo__remove"
+            onClick={handleTodoRemoval}
```

Comment:
Pass the todo id to handleTodoRemoval. Example: onClick={() => handleTodoRemoval(todo.id)}